### PR TITLE
Apisonator helper method fix and unlimited metrics case fix

### PR DESCRIPTION
**Pull request by rahulanand16nov**
This PR fixes the bug causing NoUsageFound on CacheMiss of an application that contains no limited metrics. 

Thoughts on the fix: "Since now, we know that usage_reports can be empty then there is no strict compliance for them to present. Now, the question is can there be a scenario when usage_reports are not present (even though they should be) due to an error from the SM API? Let's imagine there is, then those metrics will be treated as unlimited for the time being." 

Also, Apisonator's helper functions were not working correctly due to missing `/` or ','.

### CODE REVIEW

**Review by NomadXD — APPROVED** [review]
LGTM.

**Review by unleashed — COMMENTED** [review]
LGTM, but check out the comment.

File: cache-filter/src/filter/http.rs [hunk 1]
```diff
@@ -228,9 +226,10 @@ impl CacheFilter {
     ) -> Result<(), AuthResponseError> {
         // Form application struct from the response
         let mut state = HashMap::new();
-        let reports = response
-            .usage_reports()
-            .ok_or(AuthResponseError::UsageNotFound)?;
+        let mut reports = &Vec::<threescalers::response::UsageReport>::new();
+        if response.usage_reports().is_some() {
+            reports = response.usage_reports().unwrap();
+        }
```
[the remark this answers is not in the document]
Comment (rahulanand16nov) [reply cache-filter/src/filter/http.rs]: Changes to the for-loop is more simpler and flexible solution, I think. Only problem I faced with that is Clippy suggests us to use `if let Some(reports) = response.usage_reports() { for usage in reports ...}`; which I suppressed using `#[allow(clippy::for_loops_over_fallibles)]`

[the remark this answers is not in the document]
Comment (rahulanand16nov) [reply cache-filter/src/filter/http.rs]: Update: I just checked that `for t in Some(vec![10,20,30])` will have `t` as type of `Vec<{Integer}>` which is not what we want here.

**Review by unleashed — APPROVED** [review]

**Pull request merged**